### PR TITLE
Refactor or conditions in DoNotUseInsecureDtdProcessingAnalyzer

### LINKED ISSUE #5313: DoNotUseInsecureDtdProcessingAnalyzer is using one bar OR
I have noticed that there are a couple of places in `DoNotUseInsecureDtdProcessing` where we use the one bar or `|` (not sure what's the official name). Which forces the evaluation of both sides, I am wondering if there is any specific thing forcing this behavior?

Here are the examples:
- https://github.com/dotnet/roslyn-analyzers/blob/main/src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/DoNotUseInsecureDtdProcessing.cs#L203
- https://github.com/dotnet/roslyn-analyzers/blob/main/src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/DoNotUseInsecureDtdProcessing.cs#L212
- https://github.com/dotnet/roslyn-analyzers/blob/main/src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/DoNotUseInsecureDtdProcessing.cs#L213

Ping @dotpaul

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #5313

### CODE REVIEW

**Comment by Evangelink** [comment]
Ping @dotpaul

File: src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/DoNotUseInsecureDtdProcessing.cs [hunk 1]
```diff
@@ -200,7 +200,7 @@ public void AnalyzeOperationBlock(OperationBlockAnalysisContext context)
                 {
                     XmlDocumentEnvironment env = p.Value;
 
-                    if (!(env.IsXmlResolverSet | env.IsSecureResolver))
+                    if (!env.IsXmlResolverSet && !env.IsSecureResolver)
```
Comment (Evangelink) [inline src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/DoNotUseInsecureDtdProcessing.cs:203]: I just thought I could apply the surrounding not.

**Comment by dotpaul** [comment]
Would you mind retargeting the `2.9.x` branch please?

**Review by Youssef1313 — APPROVED** [review]

**Comment by Evangelink** [comment]
@dotpaul Could you retrigger the build please? Also I have done the retargeting.

**Review by dotpaul — APPROVED** [review]
Thank you, @Evangelink ! And thank you for the reminder. :smile:

**Pull request merged**